### PR TITLE
Suggestions from tom--

### PR DESCRIPTION
**Pull request by tom--**
I promised a PR here https://forum.yiiframework.com/t/yii-values-and-goals/123512/13

### CODE REVIEW

File: 001-yii-values.md [hunk 1]
```diff
@@ -2,75 +2,70 @@
...
-- Practical and helpful tools and community.
+The Yii project aims to build and maintain _practical_ and _helpful_ tools and community.
 
 ## Values
 
-We share some common values that serve the goal.
+The values we express in our work support the goal. We try to
 
-- Being Practical
+- Be Practical
```
Comment (tom--) [inline 001-yii-values.md]: I set the things in _imperative voice_ where it made sense.

Btw, I also think it's the best way to write code descriptions and comments. It remove the subject of the sentence. The comment/docblock description states in the form of a command and the following code will comply.

```php
/** Return the subsequent Foo for the given Bar */
function getFollowingFoo(Bar $bar): Foo
{}
```

(Btw: If you write a good description and choose good names for function and params, and use type hints, `@param` and `@return`  tags often become redundant. So remove them to be DRY.)

File: 001-yii-values.md [hunk 2]
```diff
@@ -2,75 +2,70 @@
...
+- Experience is at least as useful as theory in guiding design.
+- Context is critical in determining appropriateness.
 
-## Being Helpful
+## Be Helpful
 
-It applies both to the technical decisions and to the community. Community-wise we value any high quality contribution, be it a
-pull request or an elaborative answer at the forum.
+This value guides technical decisions as well as community activity. Consideration of how software features help users should guide development. Community-wise, we value any helpful contribution, be it a
+pull request or an answer in the forum.
```
Comment (cebe) [inline 001-yii-values.md]: 👍 for replacing "high quality" with "helpful"

File: 001-yii-values.md [hunk 3]
```diff
@@ -2,75 +2,70 @@
...
+- Loud fails are better than silent fails or masking errors.
+- Explicit casts are better than PHP type juggling, type hints are good, `declare(strict_types=1)` when using scalar type hints.
 - No magic unless necessary.
-- Loud fails are preferred to silent fails.
 
-### Consistency
+### Be consistent
 
-We try being consistent in everything:
+We try to be consistent:
```
Comment (cebe) [inline 001-yii-values.md]:
```suggestion
We try to be consistent in:
```

File: 001-yii-values.md [hunk 4]
```diff
@@ -2,75 +2,70 @@
...
+### Be Explicit
 
-For example:
+We prefer explicit, obvious solutions and code. Solutions should clearly express exactly what they do. For example:
 
-- Straightforward solution is better than over-abstracted one. That is not an excuse not to abstract though.
-- Typecasting is better than type coercion, type-hinting is better than no type hinting.
+- A straightforward solution is better than an abstract one unless the abstraction has practical purpose.
+- Loud fails are better than silent fails or masking errors.
+- Explicit casts are better than PHP type juggling, type hints are good, `declare(strict_types=1)` when using scalar type hints.
```
[the remark this answers is not in the document]
Comment (samdark) [reply 001-yii-values.md]:
```suggestion
- Explicit casts are better than PHP type juggling, type hints are good.
```

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**